proposed/local-sdk-global-json.md

```diff
@@ -0,0 +1,112 @@
+- [VS code](https://github.com/dotnet/vscode-csharp/issues/6471) having to adjust to consider local directories for SDK because our resolver can't find them.
+
+## Detailed Design
+The global.json file will support two new properties under the `sdk` object:
+
+- `"additionalPaths"`: this is a list of paths that the host resolver should consider when looking for compatible SDKs. Relative paths will be interpreted relative to the global.json file.
+- `"additionalPathsOnly"`: when `true` the resolver will _only_ consider paths in `additionalPaths`. It will not consider any machine wide locations (unless they are specified in `additionalPaths`). The default for this property is `false`.
+
+The `additionalPaths` works similar to how multi-level lookup works. It adds additional locations that the host resolver should consider when trying to resolve a compatible .NET SDK. For example:
+
+```json
+{
+    "sdk": {
+        "additionalPaths": [ ".dotnet" ],
+    }
+}
+```
+
+In this configuration the host resolver would find a compatible SDK, if it exists in `.dotnet` or a machine wide location. The host resolver will consider the `additionalPaths` in the order they are defined and will stop at the first match. If none of the locations have a matching SDK then
+the host resolver will fall back to the existing SDK resolution strategy: `%PATH%` followed by machine wide installations.
+
+This lookup will stop on the first match which means it won't necessarily find the best match. Consider a scenario with a global.json that has:
+
+```json
+{
+  "sdk": {
+    "additionalPaths": [ ".dotnet" ],
+    "version": "7.0.200",
+    "rollForward": "latestFeature"
+  }
+}
+```
+
+In a scenario where the `.dotnet` directory had 7.0.200 SDK but there was a machine wide install of 7.0.300 SDK, the host resolver would pick 7.0.200 out of `.dotnet`. That location is considered first, it has a matching SDK and hence discovery stops there.
+
+The values in the `additionalPaths` property can be a relative or absolute path. When a relative path is used it will be resolved relative to the location of global.json. These values also support the following substitutions:
+
+- `"$user$"`: this matches the user-local installation point of .NET SDK for the current operating system: `%LocalAppData%\Microsoft\dotnet` on Windows and `$HOME/.dotnet` on Linux/macOS.
+- `"$machine$"`: this matches the machine installation point of .NET for the current operating system.
```

Comment:
> Linux systems with the source-built packages installed since those have made different decisions about where to install to (and they set DOTNET_ROOT so that the rest of the system is informed about this change).

The distro provided source-built packages should not be setting DOTNET_ROOT env variable. They should be writing the system wide dotnet location under `/etc/dotnet`.

If there are any distro provided source-built packages that set DOTNET_ROOT env variables, we should get them to fix that and write under `/etc/dotnet` instead.

Reply:
I'll set up a new install on Ubuntu today and verify if this is still the case. If not then I think there's a clear semantic: on Linux/MacOS, `$machine$` should be the location written to `/etc/dotnet`. Sound good?

Reply:
The `$machine` should be equivalent to running the algorithm described in https://github.com/dotnet/designs/blob/main/accepted/2021/install-location-per-architecture.md (and in the previous https://github.com/dotnet/designs/blob/main/accepted/2020/install-locations.md). The algorithm is a bit more complicated than just looking into `/etc/dotnet` although it's true that on most systems that's what it should end up doing.

Reply:
Updated to specify it should line up with our install locations.